tools/accuracy_checker/accuracy_checker/annotation_converters/action_recognition.py

```diff
@@ -83,85 +89,126 @@ def convert(self, check_content=False, progress_callback=None, progress_interval
-                else len(list(video_path.glob('*.{}'.format(data_ext))))
-            )
-            if n_frames <= 0:
-                continue
-
-            begin_t = 1
-            end_t = n_frames
-            sample = {
-                'video': video_path,
-                'video_name': video_name,
-                'segment': [begin_t, end_t],
-                'n_frames': n_frames,
-                'video_id': video_name,
-                'label': class_to_idx[annotation['label']]
-            }
-
-            videos.append(sample)
-            if self.num_samples and len(videos) == self.num_samples:
-                break
-
+        videos = self.get_videos(video_names, annotations, class_to_idx, data_dir, data_ext)
         videos = sorted(videos, key=lambda v: v['video_id'].split('/')[-1])
 
         clips = []
         for video in videos:
-            for clip in self.get_clips(video, self.clips_per_video, self.clip_duration, self.temporal_stride, data_ext):
-                clips.append(clip)
+            clips.extend(self.get_clips(video, self.clips_per_video,
+                                        self.clip_duration, self.temporal_stride, data_ext))
 
         annotations = []
         num_iterations = len(clips)
         content_errors = None if not check_content else []
         for clip_idx, clip in enumerate(clips):
             if progress_callback is not None and clip_idx % progress_interval:
                 progress_callback(clip_idx * 100 / num_iterations)
-            identifier = ClipIdentifier(clip['video_name'], clip_idx, clip['frames'])
+            identifier = []
+            for ext in data_ext:
+                identifier.append(ClipIdentifier(clip['video_name'], clip_idx, clip['frames_{}'.format(ext)]))
             if check_content:
-                content_errors.extend([
-                    '{}: does not exist'.format(self.data_dir / frame)
-                    for frame in clip['frames'] if not check_file_existence(self.data_dir / frame)
-                ])
+                for ext, dir_ in zip(data_ext, data_dir):
+                    content_errors.extend([
+                        '{}: does not exist'.format(dir_ / frame)
+                        for frame in clip['frames_{}'.format(ext)] if not check_file_existence(dir_ / frame)
+                    ])
+            if len(identifier) == 1:
+                identifier = identifier[0]
+
             annotations.append(ClassificationAnnotation(identifier, clip['label']))
 
         return ConverterReturn(annotations, {'label_map': label_map}, content_errors)
 
-    @staticmethod
-    def get_clips(video, clips_per_video, clip_duration, temporal_stride=1, file_ext='jpg'):
-        shift = int(file_ext == 'npy')
-        num_frames = video['n_frames'] - shift
-        clip_duration *= temporal_stride
-
-        if clips_per_video == 0:
-            step = clip_duration
+    def get_ext_and_dir(self):
```

Comment:
these conditions can not be true in the same time
it means that you can just call return after check in all cases
```suggestion
    def get_ext_and_dir(self):
        if self.numpy_input and not self.image_input:
            return ['npy'], [self.data_dir / self.numpy_subdir if self.numpy_subdir else self.data_dir]
        if self.image_input and not self.numpy_input:
            return ['jpg'], [self.data_dir / self.image_subdir if self.image_subdir else self.data_dir]
        return ['jpg', 'npy'], [self.data_dir / self.image_subdir, self.data_dir / self.numpy_subdir]


            
        
```

Reply:
BTW, do you check that in the last case you have both subdirs?